core/src/test/java/org/apache/iceberg/TestDefaultValuesParsingAndUnParsing.java

```diff
@@ -0,0 +1,111 @@
+import java.util.Locale;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.util.JsonUtil;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.apache.iceberg.types.Types.NestedField.required;
+
+@RunWith(Parameterized.class)
+public class TestDefaultValuesParsingAndUnParsing {
+
+  private final Type type;
+  private final JsonNode defaultValue;
+
+  public TestDefaultValuesParsingAndUnParsing(Type type, JsonNode defaultValue) {
+    this.type = type;
+    this.defaultValue = defaultValue;
+  }
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][] {
+        {Types.BooleanType.get(), stringToJsonNode("true")},
+        {Types.IntegerType.get(), stringToJsonNode("1")},
+        {Types.LongType.get(), stringToJsonNode("9999999")},
+        {Types.FloatType.get(), stringToJsonNode("1.23")},
+        {Types.DoubleType.get(), stringToJsonNode("123.456")},
+        {Types.DateType.get(), stringToJsonNode("\"2007-12-03\"")},
+        {Types.TimeType.get(), stringToJsonNode("\"10:15:30\"")},
+        {Types.TimestampType.withoutZone(), stringToJsonNode("\"2007-12-03T10:15:30\"")},
+        {Types.TimestampType.withZone(), stringToJsonNode("\"2007-12-03T10:15:30+01:00\"")},
+        {Types.StringType.get(), stringToJsonNode("\"foo\"")},
+        {Types.UUIDType.get(), stringToJsonNode("\"eb26bdb1-a1d8-4aa6-990e-da940875492c\"")},
+        {Types.FixedType.ofLength(2), stringToJsonNode("\"111f\"")},
+        {Types.BinaryType.get(), stringToJsonNode("\"0000ff\"")},
+        {Types.DecimalType.of(9, 2), stringToJsonNode("123.45")},
+        {Types.ListType.ofOptional(1, Types.IntegerType.get()), stringToJsonNode("[1, 2, 3]")},
+        {Types.MapType.ofOptional(1, 2, Types.IntegerType.get(), Types.StringType.get()),
+         stringToJsonNode("{\"keys\": [1, 2], \"values\": [\"foo\", \"bar\"]}")},
+        {Types.StructType.of(
+            required(1, "f1", Types.IntegerType.get(), "doc"),
+            optional(2, "f2", Types.StringType.get(), "doc")),
+         stringToJsonNode("{\"1\": 1, \"2\": \"bar\"}")}
+    });
+  }
+
+  private static JsonNode stringToJsonNode(String json) {
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      return mapper.readTree(json);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("Failed to parse: " + json + "; reason: " + e.getMessage(), e);
+    }
+  }
+
+  // serialize to json and deserialize back should return the same result
+  private static String defaultValueParseAndUnParseRoundTrip(Type type, JsonNode defaultValue)
+      throws JsonProcessingException {
+    Object javaDefaultValue = DefaultValueParser.parseDefaultFromJson(type, defaultValue);
+    String jsonDefaultValue = JsonUtil.mapper()
```

Comment:
The parser should produce and accept strings, rather than doing it here in tests.

Reply:
Got it, makes sense, refactored.